### ISSUE TITLE
component id {0} exists in solution {1} already

### ISSUE THREAD

**Issue by Brentless** [SharePoint/sp-dev-fx-webparts#620]
## Category

- [X] Question
- [ ] Bug
- [] Enhancement

## Authors

@wobba 

## Expected or Desired Behavior

I have a version of the Modern Script Editor web part (basically version 1 when it first came out), and I would like to load a second version of it.  I have tried updating the v1 model, but it hasn't worked, and I would rather load a second instance and start manually switching over the previous work.  Expectation is to have two script editor web parts in the same site with different names.

## Observed Behavior

Basically what I have done is completely switched out the GUIDs in the manifest.json and the package-solution.json for completely new values, and given it a different name in the package-solution.json.  When I load it up, I get an error in the "Apps for SharePoint" page that says:

"component id {0} exists in solution {1} already"

Is there something else that needs to be different in order for SharePoint to not view the web parts as conflicts?

## Steps to Reproduce

Listed above

**Comment by wobba** [comment]
@Brentless you can not have two versions of the same wp, and it should upgrade just fine, as the behaviour is pretty identical - only improvements. If you want two versions you need to change the component id when building, so that it’s not seen as the same web part.

**Comment by VelinGeorgiev** [comment]
Sometimes SharePoint does not clean /Lists/ComponentManifests/ properly. There might be id there belonging to different package and then you have conflict. 

Some findings here:
https://blog.velingeorgiev.com/sharepoint-spfx-component-id-0-exists-solution-1-already-error

**Comment by Brentless** [comment]
ok, i've found part of my issue here. i wasnt doing a gulp clean in my process, so it had some "ghost" entries that were making there way into the final build.

I've got another one, where I'll open another post, but now my web part works fine in regular site collections, but not modern pages in Communication sites.

**Comment by KoenZomers** [comment]
Thanks for reporting this back @Brentless. That saved me from spending a whole lot of time figuring this out myself :) Ran into the same thing.

**Comment by NarenderSingh** [comment]
Check Point No. B
https://github.com/NarenderSingh/spfx-commands

**Comment by mmohapatra-muhimbi** [comment]
One more reason for getting "component id {0} exists in solution {1} already" can be, if you have multiple components in your project like extension, webpart etc and you have used multiple features but did not specified component Ids in each feature, like
```
features": [
      {
        "title": "feature title 1",
        "description": "feature description 1",
        "id": "some guid",
        "version": "1.0.0.1",
        "componentIds": [
          "component guid"
        ]
      },
      {
        "title": "feature title 2",
        "description": "feature description 2",
        "id": "some guid",
        "version": "1.0.0.1",
        "componentIds": [
          "another component guid"
        ],
        "assets": {
          "elementManifests": [
            "AppCust/elements.xml",
            "AppCust/ClientSideInstance.xml"
          ]
        }
      }
```
If we don't specify component IDs each feature adds a entry for all components in component manifests list, so we end up with duplicate entries
